### PR TITLE
Allow Undefined Behavior in Dead Code

### PR DESCRIPTION
**Pull request by Vsevolod-Livinskij**
Add Undefined Behavior in Dead Code that can be enabled with CLI.

### CODE REVIEW

File: src/expr.cpp [hunk 1]
```diff
@@ -894,9 +894,16 @@ std::shared_ptr<Expr> ArithmeticExpr::create(std::shared_ptr<PopulateCtx> ctx) {
...
     if (ctx->getArithDepth() == 0) {
-        new_node->propagateType();
-        EvalCtx eval_ctx;
-        new_node->rebuild(eval_ctx);
+        Options &options = Options::getInstance();
+        bool allow_ub = !ctx->isTaken() &&
+                        (options.getAllowUBInDC() == OptionLevel::ALL ||
+                         (options.getAllowUBInDC() == OptionLevel::SOME &&
+                          rand_val_gen->getRandId(gen_pol->ub_in_dc_prob)));
+        if (!allow_ub) {
```
Comment (dbabokin) [inline src/expr.cpp:902]: Would be good to comment why this enables UB in the dead code.

**Comment by dbabokin** [comment]
PR description is misleading - it enables UB, but just adds the option.

File: src/expr.cpp (hunk 1 above)
Comment (Vsevolod-Livinskij) [reply src/expr.cpp:904]: Fixed

**Comment by Vsevolod-Livinskij** [comment]
> PR description is misleading - it enables UB, but just adds the option.

I've changed the description

**Pull request merged**